### ISSUE TITLE
Use can-element for guides and recipes

### ISSUE THREAD

**Issue by phillipskevin** [canjs/canjs#3570]
**tldr** [`can-element`](https://canjs.com/doc/can-element.html) allows you to create [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements) with CanJS. There are several benefits that make `can-element` preferable to `can-component` and so we should use `can-element` to teach creating reusable elements in our guides and recipes.

## Details

Custom Elements are part of the Web Components [specification](https://w3c.github.io/webcomponents/spec/custom/). Custom Elements have a few benefits over `can-component`:

- elements are automatically parsed and constructed
- elements can interact with any library or framework through standard DOM attributes and events
- styles can be isolated using shadow DOM

### Automatic construction

When a Custom Element is defined using `customElements.define('my-component', MyComponent);` the HTML parser will automatically look for `<my-component />`s in the HTML and call the `MyComponent` constructor to create the Custom Element.

Because of this, creating a custom element in a JSBin for a guide or at a meetup is as simple as adding the element to the html tab like any other element.

```html
<my-component></my-component>
```

This is much easier to explain and much harder for people to make mistakes than the current approach:

```js
var renderer = can.stache('<my-component></my-component>');
var frag = renderer({});
document.body.appendChild(frag);
```

### Interopability

Custom Elements interact through standard DOM APIs - events and attributes. This means that a `can-element` can interoperate with elements/components created by any library or framework through these APIs.

This also means you can use browser devtools to interact with `can-element`:

![components](https://user-images.githubusercontent.com/5851984/30714367-2b6b4898-9ed8-11e7-97e7-119a6909c8d2.gif)

### Shadow DOM

Custom Elements created by `can-element` are rendered to [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Shadow_DOM) by default. This means that styles for the element are completely isolated to the element.

This means that styles like
```css
p {
    color: red;
}
```

will not affect paragraphs outside the Custom Element. This can greatly simplify CSS and avoid issues with clashing styles.

## Appendix A: Browser Support

Custom Elements are currently supported in Chrome and Safari. There is a [polyfill](https://github.com/webcomponents/custom-elements/blob/master/custom-elements.min.js) available for other browsers.

**Comment by justinbmeyer** [comment]
I would like to see the `events` object change to avoid the  `{}` in `"{data} thing"` sorta stuff, especially since the element is the view model now.

Could we have something like:

```js
events: {
  connected: function(){
    setInterval(() => {
                this.time = new Date();
            }, 1000);
  },
  disconnected: function(){},
  "li click": function(){}
}
```

**Comment by matthewp** [comment]
I think it should still be possible to specify an (external) view model for these elements. The use case is I would like to put the view model in a web worker and provide a VM proxy to can-element.

**Comment by matthewp** [comment]
Epic is here: https://github.com/canjs/can-element/issues/37
